model/task/task_test.go

```diff
@@ -3496,6 +3496,23 @@ func TestAddParentDisplayTasks(t *testing.T) {
 	assert.Equal(t, dt2.Id, tasks[3].DisplayTask.Id)
 }
 
+func TestSetCheckRunId(t *testing.T) {
+	require.NoError(t, db.Clear(Collection))
+	t1 := &Task{
+		Id: "t1",
+	}
+
+	assert.NoError(t, t1.Insert())
+	assert.NoError(t, t1.SetCheckRunId(12345))
+
+	var err error
+	t1, err = FindOneId(t1.Id)
+	assert.NoError(t, err)
+
+	assert.Equal(t, t1.CheckRunId, utility.ToIntPtr(12345))
```

Comment:
nit: it's usually considered better to compare values (which seems like what you're trying to do) rather than pointers because pointers are just meaningless addresses, so if go so happened to decide to give them the same address, it's not something we should rely on. This should instead be:
```go
assert.Equal(t, 12345, utility.FromIntPtr(t1.CheckRunId))
```
(also note the flipped arguments because the first value is always the expected value and the second is the actual value).